webapp/channels/src/components/new_search/search_box_suggestions.tsx

```diff
@@ -0,0 +1,164 @@
+}
+
+const SearchSuggestions = ({searchType, searchTerms, setSearchTerms, getCaretPosition, suggestionsHeader, providerResults, selectedOption, setSelectedOption, focus, onSearch}: Props) => {
+    const license = useSelector(getLicense);
+    const updateSearchValue = useCallback((value: string, matchedPretext: string) => {
+        const caretPosition = getCaretPosition();
+        const extraSpace = caretPosition === searchTerms.length ? ' ' : '';
+        setSearchTerms(searchTerms.slice(0, caretPosition).replace(new RegExp(matchedPretext + '$'), '') + value + extraSpace + searchTerms.slice(caretPosition));
+        focus((caretPosition + value.length + 1) - matchedPretext.length);
+    }, [searchTerms, setSearchTerms, focus, getCaretPosition]);
+
+    const runSearch = useCallback((searchTerms: string) => {
+        onSearch(searchType, searchTerms);
+    }, [onSearch, searchType]);
+
+    let SearchPluginSuggestions = useSelector((state: GlobalState) => state.plugins.components.SearchSuggestions) || [];
+    if (license.IsLicensed !== 'true') {
+        SearchPluginSuggestions = [];
+    }
+
+    const generateLabel = (item: any) => {
+        let label = '';
+        if (item.username) {
+            label = item.username;
+            if ((item.first_name || item.last_name) && item.nickname) {
+                label += ` ${item.first_name} ${item.last_name} ${item.nickname}`;
+            } else if (item.nickname) {
+                label += ` ${item.nickname}`;
+            } else if (item.first_name || item.last_name) {
+                label += ` ${item.first_name} ${item.last_name}`;
+            }
+        } else if (item.type === 'D' || item.type === 'G') {
+            label = item.display_name;
+        } else if (item.type === 'P' || item.type === 'O') {
+            label = item.name;
+        } else if (item.emoji) {
+            label = item.name;
+        }
+
+        if (label) {
+            label = label.toLowerCase();
+        }
+        return label;
+    };
+
+    if (searchType === '' || searchType === 'messages' || searchType === 'files') {
+        if (!providerResults) {
+            return null;
+        }
+        return (
+            <SuggestionsBody>
+                <SuggestionsHeader>{suggestionsHeader}</SuggestionsHeader>
+                {providerResults.items.map((item, idx) => {
+                    if (!providerResults.component) {
+                        return null;
+                    }
+                    if (idx !== selectedOption) {
+                        return null;
+                    }
+                    return (
+                        <div
+                            aria-live='polite'
+                            role='alert'
+                            className='sr-only'
+                            key={providerResults.terms[idx]}
+                        >
+                            {idx === selectedOption ? generateLabel(item) : ''}
```

Comment:
This component not being mounted when a plugin is providing results does also make me wonder a bit about the accessibility of plugin search, but I don't know how much we've really focused on that for plugins

Reply:
The idea here is that the plugin should have its own code for accesibility.

Reply:
the `idx === selectedOption` is fixed now, and defining the function I think is right in this context because putting that directly there, taking into account the that this is a lot of ifs, looks more clean to me.